src/components/Progress/Steps.tsx

```diff
@@ -0,0 +1,147 @@
+        const itemWidth: number = Math.round(
+            Math.floor(flexContainerWidth / steps) -
+                stepItemMarginOffset +
+                Math.ceil(stepItemMarginOffset / steps)
+        );
+
+        // Calculates the percent width of each item
+        setCalculatedWidth(`${(itemWidth / flexContainerWidth) * 100}%`);
+    }, [steps]);
+
+    const updateLayout = (): void => {
+        // Early exit if there is no ref available yet. The DOM has yet to initialize
+        // and its not possible to calculate positions.
+        if (!valueLabelRef?.current || !showValueLabel) {
+            return;
+        }
+
+        // Hide the min/max labels if the value labels would collide.
+        let progressBarOffset: number;
+        const valueLabelOffset: number = valueLabelRef.current.offsetWidth;
+
+        let showMaxLabel: boolean;
+        let showMinLabel: boolean;
+
+        if (directionConfig === 'rtl') {
+            progressBarOffset = progressBgRef.current.offsetWidth;
+
+            showMaxLabel =
+                showLabels &&
+                valueLabelRef.current.getBoundingClientRect().left >
+                    maxLabelRef.current.getBoundingClientRect().right;
+            maxLabelRef.current.style.opacity = showMaxLabel ? '1' : '0';
+
+            showMinLabel =
+                showLabels &&
+                valueLabelRef.current.getBoundingClientRect().left +
+                    valueLabelOffset <
+                    minLabelRef.current.getBoundingClientRect().left;
+            minLabelRef.current.style.opacity = showMinLabel ? '1' : '0';
+
+            valueLabelRef.current.style.right = `${
+                progressBarOffset - valueLabelOffset / 2
+            }px`;
+            valueLabelRef.current.style.left = 'unset';
+        } else {
+            progressBarOffset =
+                progressBgRef.current.getBoundingClientRect().right;
+            showMaxLabel =
+                showLabels &&
+                valueLabelRef.current.getBoundingClientRect().left +
+                    valueLabelOffset <
+                    maxLabelRef.current.getBoundingClientRect().left;
+            maxLabelRef.current.style.opacity = showMaxLabel ? '1' : '0';
+
+            showMinLabel =
+                showLabels &&
+                valueLabelRef.current.getBoundingClientRect().left >
+                    minLabelRef.current.getBoundingClientRect().right;
+            minLabelRef.current.style.opacity = showMinLabel ? '1' : '0';
+
+            valueLabelRef.current.style.left = `${
+                progressBarOffset - valueLabelOffset
+            }px`;
+            valueLabelRef.current.style.right = 'unset';
+        }
+    };
+
+    useLayoutEffect(() => {
+        updateLayout();
+    }, [showLabels, showValueLabel]);
+
+    return (
+        <ResizeObserver onResize={updateLayout}>
+            <div ref={flexContainerRef} className={styles.progressStepsOuter}>
```

Comment:
are these accessable ? Same question for all the other components

Reply:
The text content is accessible. The visual representations would only need tab indices and aria label props. I'll add these.

Reply:
To answer your question. The Octuple component library is accessible. However, there will be bugs (as always) that are device specific - iOS, Android, Windows, MacOS, Ubuntu, etc... all use different accessible technologies and the test matrix is massive due to unique versions, browsers and more. Much too great to cover 100% in 4 months time. I will say this however, the last product I worked on was a gleaming example of A11y support for hundreds of millions of users (MAU/DAU). I'm striving to apply those best practices here, though it is a living effort and will never be 'complete'.

Reply:
Now if we could migrate vscode app to use Layout, Grid, and Stack to support A11y regions that would be dope.

Reply:
No A11y violations found using the addon.